File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        k=0
        
        for i in nums:
            if i!=0:
                nums[k]=i
                k+=1
        for i in range(k,len(nums)):
            nums[i]=0



        
        

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int slow = 0;//checks for the non zero element

        for(int i = 0; i<nums.size(); i++){//scans through entire array
            if(nums[i]!=0){
                //swap slow and non zero
                swap(nums[slow],nums[i]);
                //forwards slow
                slow++;
            }

        }

    }
};